File: scripts/audit_and_update_contractors.py
from supabase import create_client
import json
import os

# Initialize Supabase
url = os.environ.get('NEXT_PUBLIC_SUPABASE_URL')
key = os.environ.get('NEXT_PUBLIC_SUPABASE_ANON_KEY')
supabase = create_client(url, key)

def get_specialty_from_business_name(name):
    name_lower = name.lower()
    if 'roof' in name_lower: return ['Roofing']
    if 'electric' in name_lower: return ['Electrical']
    if 'plumb' in name_lower: return ['Plumbing']
    if 'hvac' in name_lower or 'heating' in name_lower or 'air' in name_lower: return ['HVAC']
    if 'handyman' in name_lower: return ['Handyman']
    return ['General Contractor']

def run_audit():
    # Fetch all with empty specialties
    response = supabase.from_('contractors').select('id, business_name, specialties').is_('specialties', 'null').execute()
    contractors = response.data
    
    print(f"Auditing {len(contractors)} contractors...")
    
    for c in contractors:
        new_specialty = get_specialty_from_business_name(c['business_name'])
        supabase.from_('contractors').update({'specialties': new_specialty}).eq('id', c['id']).execute()
        print(f"Updated {c['business_name']} to {new_specialty}")

run_audit()

File: scripts/audit_and_update_contractors.js
const { createClient } = require('@supabase/supabase-js');
const fs = require('fs');
const envContent = fs.readFileSync('/data/.openclaw/workspace/prolink-ezly/.env', 'utf8');
const env = envContent.split('\n').reduce((acc, line) => {
    const [key, value] = line.split('=');
    if (key) acc[key.trim()] = value ? value.trim() : '';
    return acc;
}, {});
const supabase = createClient(env.NEXT_PUBLIC_SUPABASE_URL, env.NEXT_PUBLIC_SUPABASE_ANON_KEY);

function getSpecialty(name) {
    const n = name.toLowerCase();
    if (n.includes('roof')) return ['Roofing'];
    if (n.includes('electric')) return ['Electrical'];
    if (n.includes('plumb')) return ['Plumbing'];
    if (n.includes('hvac') || n.includes('heating')) return ['HVAC'];
    if (n.includes('handyman')) return ['Handyman'];
    return ['General Contractor'];
}

async function audit() {
    const { data: contractors } = await supabase
        .from('contractors')
        .select('id, business_name')
        .or('specialties.is.null,specialties.eq.{}'); // Matches null or empty array

    if (!contractors) return;

    for (const c of contractors) {
        const specialty = getSpecialty(c.business_name);
        await supabase.from('contractors').update({ specialties: specialty }).eq('id', c.id);
        console.log(`Updated ${c.business_name} -> ${specialty}`);
    }
}
audit();
